### ISSUE TITLE
Selectively disabling certain packages from being handled by Hunter

### ISSUE THREAD

**Issue by themarpe** [cpp-pm/hunter#557]
* I've read [Brief overview section](https://hunter.readthedocs.io/en/latest/overview.html) and do understand basic concepts. **[Yes]**
* I've read [F.A.Q. section](https://hunter.readthedocs.io/en/latest/faq.html) and haven't found an answer to my question. **[Yes]**
* I've read [Code of Conduct](https://github.com/cpp-pm/hunter/blob/master/.github/CODE_OF_CONDUCT.md), I promise to be polite and will do my best at being constructive. **[Yes]**

---

Should we add in support to selectively skip certain packages being handled by Hunter, as is done by turning off whole Hunter with `HUNTER_ENABLED=OFF` currently?

Something along the lines of `HUNTER_SKIP_PACKAGE_[package_name]` specifying if `hunter_add_package` for X package should do nothing, resulting in `find_package` to search  from the system/...,

Eg: `-D HUNTER_SKIP_PACKAGE_nlohmann_json=ON`

Not sure how deep the changes would have to be to support such use case and/or if this would make sense.

Main usecase I see is selecting certain dependency from system for means of packaging (eg relying on nlohmann json from system package manager, so dependencies are kept inline, etc...)

Thoughts?

**Comment by NeroBurner** [comment]
Although I understand the wish for this feature it is a can of worms resulting in many many headaches

https://hunter.readthedocs.io/en/latest/faq/interaction-with-other-package-managers.html

**Comment by themarpe** [comment]
I agree, but not in cases where Hunter is used also as a private package management system. Eg. getting certain dependencies that an executable/library relies on, but only internally. In that case I think it makes sense to then use Hunter for ease of use (eg requiring no extra user work to supply such dependencies), but still gives the freedom to switch out to an external library, for others, when required. (eg to not have conflicts, when creating a deb file, etc...)

**Comment by traversaro** [comment]
Hello @NeroBurner ! We were discussing about this issue in https://github.com/luxonis/depthai-core/issues/1021#issuecomment-2110307466, and I guess it is just easier to ask you: are you interested in reviewing a proposal for this feature or not? Thanks in advance!

**Comment by NeroBurner** [comment]
yes, if the feature is well tested and invalidates the cache to prevent issues mentioned in the FAQ

**Comment by traversaro** [comment]
Thanks! I implemented the super barebone logic in https://github.com/traversaro/hunter/tree/fix557, suggestions for how to handle disabling cache when  `HUNTER_SKIP_PACKAGE_<packageName>` is used are welcome. As this is extremely advanced option (that in general is discouraged) I wonder if it could make sense to just permit to use only when caching is explicitly disabled (but I do not know if there is any way to disable all caching).
